lib/posix/options/CMakeLists.txt

```diff
@@ -75,10 +75,6 @@ if (NOT CONFIG_TC_PROVIDES_POSIX_FILE_SYSTEM_R)
   zephyr_library_sources_ifdef(CONFIG_POSIX_FILE_SYSTEM_R file_system_r.c)
 endif()
 
-zephyr_library_sources_ifdef(CONFIG_POSIX_FSYNC fsync.c)
-zephyr_library_sources_ifdef(CONFIG_POSIX_MEMLOCK mlockall.c)
-zephyr_library_sources_ifdef(CONFIG_POSIX_MEMLOCK_RANGE mlock.c)
-
 if (NOT CONFIG_TC_PROVIDES_POSIX_MEMORY_PROTECTION)
 zephyr_library_sources_ifdef(CONFIG_POSIX_MEMORY_PROTECTION mprotect.c)
```

Comment:
```suggestion
if(NOT CONFIG_TC_PROVIDES_POSIX_MEMORY_PROTECTION)
  zephyr_library_sources_ifdef(CONFIG_POSIX_MEMORY_PROTECTION mprotect.c)
```
(for a future PR)

Reply:
Good catch - we should probably have a cmake linter in GH actions

Reply:
@nordicjm - created https://github.com/zephyrproject-rtos/zephyr/issues/84876